.github/workflows/mypy.yml

```diff
@@ -0,0 +1,46 @@
+name: MyPy
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 'release/*'
+  workflow_dispatch:
+
+# Cancel previous running jobs on the same branch in case of new pushes
+concurrency:
+  group: mypy-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install MyPy
+        run: |
+          pip install mypy
+          pip install ./sdk_commons ./template ./opyn ./friktion ./ribbon
+
+      - name: Run MyPy on Template
+        if: always()
+        run: mypy template --exclude abis --exclude tests
+
+      - name: Run MyPy on Ribbon
```

Comment:
We should probably run it on the single folder if there are changes, otherwise not.

Template should always run instead.

Reply:
It was more of a suggestion to use one or the other.

For the record, reading around, you could do two steps like:
1. generate the matrix
2. execute the matrix generated

Example:
https://purple.telstra.com/blog/github-action-test-matrices-part2

But we can keep that experiment for another time :)